1286 tuning stage 2 of candidate results projecting to summary dtos instead of entities

camerojo opened the pull request.

camerojo committed:
Add summary flag allowing improved performance for displaying candidate results by only fetching summary fields.

Added new CandidateBuilderSelector.selectMinimalBuilder method - replacing the old boolean call on selectBuilder.

EAGER fetch some many to one Candidate fields that reduce total database accesses.

camerojo committed:
Remove some Intellij warnings around lack of typing.

camerojo committed:
Doc only

camerojo committed:
First start at non entity dtos

camerojo committed:
Rename to doSearchCandidateSummaries

Commit:
Add search timings

camerojo committed:
Merge branch 'staging' into 1286-tuning-stage-2-of-candidate-results-projecting-to-summary-dtos-instead-of-entities

# Conflicts:
#	server/src/main/java/org/tctalent/server/api/admin/CandidateAdminApi.java
#	server/src/main/java/org/tctalent/server/api/admin/CandidateBuilderSelector.java
#	server/src/main/java/org/tctalent/server/api/admin/SavedListCandidateAdminApi.java
#	server/src/main/java/org/tctalent/server/api/admin/SavedSearchCandidateAdminApi.java
#	server/src/main/java/org/tctalent/server/model/db/Candidate.java
#	server/src/main/java/org/tctalent/server/model/db/TaskDtoHelper.java
#	server/src/main/java/org/tctalent/server/request/candidate/SearchCandidateRequest.java
#	server/src/main/java/org/tctalent/server/service/db/SavedSearchService.java
#	server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java
#	ui/admin-portal/src/app/components/util/cv-icon/cv-icon.component.ts

camerojo committed:
javax to jakarta

Fix the doc

camerojo committed:
Get basic DTO stuff working

camerojo committed:
Embed dtos search into standard search logic and time it.

camerojo committed:
Modify processing path to use dtos instead of entities

camerojo committed:
Fill out CandidateDto and CandJobExDto

camerojo committed:
First cut at support for User and Partner

camerojo committed:
Restructure to use new repository.db.read package

camerojo committed:
Fill in sql builder and annotation classes.

Update CandidateReadDao to use them and the new CandidateReadMapper

camerojo committed:
Create sqlBuilder test

Move Job experience dto

camerojo committed:
Get simple test working

camerojo committed:
Convert all returned data into json.

Optional mapping of all unannotated attributes

camerojo committed:
Add Partner dto to check nesting.

Add different User attributes.

camerojo committed:
Added SqlIgnore

Simplified annotation params

camerojo committed:
Doc only

camerojo committed:
Doc only

camerojo committed:
Need to have id on candidate dtos

camerojo committed:
Make SqlColumn default name to field name

camerojo committed:
Add Fast search check box to front end.

Duplicate method on back end for fast search

camerojo committed:
Ignore computed fields rank and selected

camerojo committed:
Doc only

camerojo committed:
Doc only

camerojo committed:
Add some read dtos

camerojo committed:
Add missing Dtos

camerojo committed:
Fixed links

Added SalesforceJobOppReadDto

camerojo committed:
Rank is a number

camerojo committed:
Camel to snake case handles numbers

camerojo committed:
Get something working by removing problems - marked with dtos

camerojo committed:
Changed version which chunks up calls to jsonb build function to get around its 100 argument limit.

camerojo committed:
Fixed some mistaken Booleans

Add missing Dtos

camerojo committed:
Remove commented out code

camerojo committed:
Change CandidateDtoService to return a Map of ids to DTOs.

camerojo committed:
Change sql builder to return JSON in a single field called json.

camerojo committed:
Add json caching tables

camerojo committed:
Merge branch 'Flyways' into 1286-tuning-stage-2-of-candidate-results-projecting-to-summary-dtos-instead-of-entities

camerojo committed:
Start building out CandidateDtoServiceImpl

camerojo committed:
Add in CandidateVersionDao

camerojo committed:
All classes in

camerojo committed:
Set ttl to null for now.

camerojo committed:
Put CandidateRedisCache into logic.

Replace logging with logBuilder

Rename some variables

camerojo committed:
Doc only

camerojo committed:
Doc only

camerojo committed:
Remove unneeded timing code for fetching dtos

camerojo committed:
Support candidate selected property in CandidateReadDto

camerojo committed:
Change Strings to enum types

camerojo committed:
Fix trigger bug when updating a candidate field.

camerojo committed:
Merge branch 'Flyways' into 1286-tuning-stage-2-of-candidate-results-projecting-to-summary-dtos-instead-of-entities

camerojo committed:
Rename method for consistency

camerojo committed:
Implement getUrl

camerojo committed:
New SqlJsonBuilder - should manage nested 1-many relations

sadatmalik commented on server/src/main/java/org/tctalent/server/service/db/impl/CandidateDtoServiceImpl.java:
```diff
@@ -0,0 +1,235 @@
+                    + ", versions=" + versions)
+                .logError();
+            throw new NoSuchObjectException(
+                "Candidates not found with ids: " + badIds
```
Think this should log unprocessedIds, not badIds?

camerojo replied:
Nice catch! Thanks!

sadatmalik commented:
Great stuff!

Approved with just one tiny comment

camerojo committed:
Display unprocessedIds

camerojo committed:
Better operation of search variants.

Note that the variants are now only enabled when a loaded saved search has been modified.
